### PR TITLE
blog1@a-star

### PR DESCRIPTION
**Pull request by A-star-ayush**
Introductory blog for GSOC 2017

### CODE REVIEW

File: content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+I am a profound Linux user. I have gained formal education on the subject while pursuing the Redhat certifications. I have always been fascinated with the intricacies behind the scene, the internal working of an operating system which makes you believe that the trivial program of ‘Hello world’ is actually trivial.<br><br>
+</p>
+
+<h3>Project Motivation</h3>
+
+<p>TCP and IP together form the most ubiquitous pair of protocols used in the implementation of the internet stack. It’s really important to get them working at full throttle. When I was going through the list of issues at Trac, a lot of them complained about the network:<br>
+
+<ul>
+<li>#1958, #12815 - slow speeds during file exchange using ftp, scp, sftp </li>
```
Comment (pulkomandy) [inline content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md]: Hyperlinks to Trac tickets sound like a nice idea here?

File: content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+<li>#10747, #11066, #1143 - network related KDLs </li>
+<li>#1073 - requesting implementation of path MTU discovery  </li>
+<li>#2967 - bug with getpeername() that succeeds on unconnected sockets  </li>
+<li>#1994 - requesting implementation of urgent data for TCP  </li>
+</ul>
+<br>
+and more ...<br>
+
+RFC 7414 is a roadmap for implementors of TCP and I shall be following it closely, implementing features not already present or not fully implemented.
```
Comment (pulkomandy) [inline content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md]: Hyperlink to RFC?

File: content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+</p><br>
+
+<h3>Current state</h3>
+
+<p>The code for TCP is found under src/add-ons/kernel/network/protocols/tcp. It provides for all the mainframe syscalls like: bind, listen, sendmsg, etc. Needless to say, all the salient features of tcp like checksum, window, ack, fin are well implemented.</p>
```
Comment (pulkomandy) [inline content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md]: Hyperlink to cgit?

File: content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+<p>During summer I will be implementing each section of rfc 7414 one after the other. There are majorly 3 sections and I will have 3 months to work on them:</p>
+
+<ul>
+<li>Section 2: describes the very fundamental functionalities of tcp. </li>
+<li>Section 3: Strongly Encouraged Enhancements - describes recommended TCP modifications that improve performance and security</li>
+<li>Section 4: Experimental Extensions - describes those extensions that are not widely but are gaining popularity </li>
+</ul><br>
+
+To more details on the timeline, you can refer to my proposal:
+https://docs.google.com/document/d/1-FlypLFN2c1OOSkwh9N72uuQWMbY3hHiTwhFSqPaJbM/edit
```
Comment (pulkomandy) [inline content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md]: Hyperlink?

File: content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md (hunk 1 above)
Comment (A-star-ayush) [inline content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md]: Done

File: content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md (hunk 2 above)
Comment (A-star-ayush) [inline content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md]: Done

File: content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md (hunk 3 above)
Comment (A-star-ayush) [inline content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md]: Done

File: content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md (hunk 4 above)
Comment (A-star-ayush) [inline content/blog/a-star/2017-05-10_[gsoc_2017]_tcp_optimization_and_fine_tuning.md]: Done

**Pull request merged**